### PR TITLE
[MIN-208] Add integration test

### PR DESCRIPTION
**Pull request by dev-vadim**
**Description:**

Describe in brief, what was done.

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [ ] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

**Review by lut1k — APPROVED** [review]

File: pallets/integration-tests/src/tests/mnt_token_tests.rs [hunk 1]
```diff
@@ -175,4 +175,158 @@ mod tests {
...
+					MDOT,
+					Currencies::free_balance(MDOT, &ALICE)
+				));
+				// Return the same proportions (1:2) eventually
+				assert_ok!(MinterestProtocol::transfer_wrapped(
+					bob(),
+					ALICE,
+					MDOT,
+					Currencies::free_balance(MDOT, &BOB) / 3
+				));
```
Comment (kurotychb) [inline pallets/integration-tests/src/tests/mnt_token_tests.rs:301]: This block seems useless for me. Just make one transfer what you need.

File: pallets/integration-tests/src/tests/mnt_token_tests.rs [hunk 2]
```diff
@@ -175,4 +175,158 @@ mod tests {
...
 			})
 	}
+
+	// Test MNT distribution behaviour when users are using transfer_wrapped
+	#[test]
+	fn mnt_token_supplier_distribution_when_users_transferring_tokens() {
+		ExtBuilder::default()
+			.pool_initial(DOT)
+			.pool_initial(ETH)
+			.pool_initial(BTC)
```
Comment (kurotychb) [inline pallets/integration-tests/src/tests/mnt_token_tests.rs:185]: Why do you use setting for 3 currencies but use in test only 1?

File: pallets/integration-tests/src/tests/mnt_token_tests.rs (hunk 1 above)
Comment (dev-vadim) [reply pallets/integration-tests/src/tests/mnt_token_tests.rs:301]: Just wanted to check that several transfers between users within one block don`t break anything

Comment (kurotychb) [reply pallets/integration-tests/src/tests/mnt_token_tests.rs:301]: Okay, maybe it has some point

File: pallets/integration-tests/src/tests/mnt_token_tests.rs (hunk 2 above)
Comment (dev-vadim) [reply pallets/integration-tests/src/tests/mnt_token_tests.rs]: That was a copy&paste, thanks)
Removed ETH and BTC pools

File: pallets/integration-tests/src/tests/mnt_token_tests.rs [hunk 3]
```diff
@@ -175,4 +175,148 @@ mod tests {
...
+
+				// Alice deposits DOT and enables her DOT pool as a collateral.
+				assert_ok!(MinterestProtocol::deposit_underlying(alice(), DOT, ONE_HUNDRED));
+				assert_ok!(MinterestProtocol::enable_is_collateral(alice(), DOT));
+				// Bob deposits DOT and enables her DOT pool as a collateral.
+				assert_ok!(MinterestProtocol::deposit_underlying(bob(), DOT, ONE_HUNDRED));
+				assert_ok!(MinterestProtocol::enable_is_collateral(bob(), DOT));
+				// Carol deposits DOT and enables her DOT pool as a collateral.
+				assert_ok!(MinterestProtocol::deposit_underlying(carol(), DOT, 2 * ONE_HUNDRED));
+				assert_ok!(MinterestProtocol::enable_is_collateral(carol(), DOT));
```
Comment (kurotychb) [inline pallets/integration-tests/src/tests/mnt_token_tests.rs:206]: Seems `enable_is_collateral` is not required to be in this tests. It should be enough just to deposit values.

Comment (dev-vadim) [reply pallets/integration-tests/src/tests/mnt_token_tests.rs]: Done

**Review by lut1k — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**